src/CSharp/CodeCracker/Style/StringFormatCodeFixProvider.cs

```diff
@@ -20,16 +21,24 @@ public class StringFormatCodeFixProvider : CodeFixProvider
 
         public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
 
-        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
             var diagnostic = context.Diagnostics.First();
-            var diagnosticSpan = diagnostic.Location.SourceSpan;
-            var invocation = root.FindToken(diagnosticSpan.Start).Parent.AncestorsAndSelf().OfType<InvocationExpressionSyntax>().First();
-            context.RegisterCodeFix(CodeAction.Create("Change to string interpolation", c => MakeStringInterpolationAsync(context.Document, invocation, c)), diagnostic);
+            context.RegisterCodeFix(CodeAction.Create(Resources.StringFormatCodeFixProvider_Title, c => MakeStringInterpolationAsync(context.Document, diagnostic, c)), diagnostic);
+            return Task.FromResult(0);
         }
 
-        private async Task<Document> MakeStringInterpolationAsync(Document document, InvocationExpressionSyntax invocationExpression, CancellationToken cancellationToken)
+        private async Task<Document> MakeStringInterpolationAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var invocationExpression = root.FindToken(diagnostic.Location.SourceSpan.Start).Parent.AncestorsAndSelf().OfType<InvocationExpressionSyntax>().First();
+            var newStringInterpolation = await CreateNewStringInterpolationAsync(document, invocationExpression, cancellationToken);
+            var newRoot = root.ReplaceNode(invocationExpression, newStringInterpolation);
+            var newDocument = document.WithSyntaxRoot(newRoot);
+            return newDocument;
```

Comment:
It's nothing important, but there is no need to creating variable `newDocument`, it just can return `document.WithSyntaxRoot(newRoot);`